x-pack/plugins/apm/server/routes/services.ts

```diff
@@ -39,7 +33,7 @@ export const servicesRoute = createRoute(() => ({
     const services = await getServices({
       setup,
       searchAggregatedTransactions,
-      mlAnomaliesEnvironment: environment,
+      mlAnomaliesEnvironment: setup.uiFilters.environment,
```

Comment:
This is not necessary since `setup` is already passed in, right?
```suggestion
```

Reply:
yes, i went down this rabbit hole and it touched a lot more files, so i just skipped it for now, but i could go thru and fix this as well